src/tools/clippy/clippy_lints/src/significant_drop_in_scrutinee.rs

```diff
@@ -391,6 +364,57 @@ impl<'a, 'tcx> Visitor<'tcx> for SigDropHelper<'a, 'tcx> {
                     self.has_significant_drop = true;
                 }
             }
+            ExprKind::Box(expr) => self.visit_expr(expr),
```

Comment:
Oh, sorry this wasn't clear: You can and should keep the `walk_expr`. I just meant you should list those cases explicitly here, just as 

```rust
ExprKind::MethodCall(..) | ExprKind::Box(..) | ... => walk_expr(self, expr)
```

Reply:
Ah, I see! Thank you for the clarification!